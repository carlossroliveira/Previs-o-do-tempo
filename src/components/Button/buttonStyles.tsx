import styled from 'styled-components';

export const ContainerSC = styled.div`
  display: grid;
  place-items: center;
`;
export const ButtonSC = styled.a`
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  text-decoration: none;

  color: ${({ theme }) => theme.color.primary};
  border: ${({ theme }) => theme.color.primary} 0.125em solid;

  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3),
    0 0 0.45em ${({ theme }) => theme.color.primary};
  box-shadow: inset 0 0 0.5em 0 white, 0 0 0.5em 0 white;

  &::before {
    content: '';
    position: absolute;
    pointer-events: none;
    top: 120%;
    left: 0;
    width: 100%;
    height: 100%;
    opacity: 0.7;
    filter: blur(1em);
    background-color: ${({ theme }) => theme.background.secondary};
    transform: perspective(1em) rotateX(40deg) scale(1, 0.35);
  }

  &:hover,
  &:focus {
    color: ${({ theme }) => theme.color.secondary};
    background-color: ${({ theme }) => theme.background.secondary};
    text-shadow: none;
  }

  &:hover::before,
  &:focus::before {
    opacity: 1;
  }
`;
