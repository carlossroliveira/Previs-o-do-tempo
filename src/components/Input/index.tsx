// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useMemo } from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { useMyHookApplication } from '../../Context/contextApplication/ContextTheme';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, InputSC } from './inputStyles';
import { Hourglass } from 'react-spinners-css';

export const Input = (): JSX.Element => {
  const { storage, loading, validateInput, onChangeInformation } =
    useMyHookApplication();

  const errorMessage = useMemo(() => {
    return !!validateInput;
  }, [validateInput]);

  return (
    <ContainerSC error={errorMessage}>
      <InputSC
        type="text"
        placeholder="Digite seu estado"
        value={storage}
        onChange={onChangeInformation}
      />

      {loading && <Hourglass color={'#393E46'} />}
    </ContainerSC>
  );
};
