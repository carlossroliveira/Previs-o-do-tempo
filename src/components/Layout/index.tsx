// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Header } from './part/Header';
import { ContentOne } from './part/ContentOne';
import { ContentTwo } from './part/ContentTwo';
import { Footer } from './part/Footer';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './layoutStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

export const Layout = (): JSX.Element => {
  return (
    <ContainerSC>
      <Header />
      <ContentOne />
      <ContentTwo />
      <Footer />
    </ContainerSC>
  );
};
