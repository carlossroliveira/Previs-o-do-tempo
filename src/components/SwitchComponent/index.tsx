// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useState } from 'react';
import { useMyHook } from '../hooks/theme';
import { ParagraphSC, Toggle } from './SwitchComponentStyles';

export const SwitchComponent = (): JSX.Element => {
  const { handleThemes } = useMyHook();

  const [first, setfirst] = useState<boolean>(false);

  const test = () => {
    setfirst((firstt) => !firstt);
    handleThemes();
  };

  return (
    <>
      <ParagraphSC>{!first ? 'Light' : 'Dark'}</ParagraphSC>
      <Toggle
        checked={first}
        uncheckedIcon={false}
        checkedIcon={false}
        onChange={test}
      />
    </>
  );
};
