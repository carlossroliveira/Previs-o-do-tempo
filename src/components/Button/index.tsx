// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';

export const Button = (): JSX.Element => {
  return (
    <ContainerSC>
      <ButtonSC>Buscar</ButtonSC>
    </ContainerSC>
  );
};
