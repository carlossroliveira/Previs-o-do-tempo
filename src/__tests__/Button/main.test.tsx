// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { Button } from '../../components/Button';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../utils/withThemeComponent';

const ButtonWithTheme = WithThemeComponent(Button);

describe('Button Testing', () => {
  xit('Should render the button component', () => {
    render(<ButtonWithTheme />);
    expect(screen.getByText(/buscar/i)).toBeInTheDocument();
  });
});
