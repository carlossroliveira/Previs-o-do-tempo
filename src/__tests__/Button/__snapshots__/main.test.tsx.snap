// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: grid;
  place-items: center;
}

.c1 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c1::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c1:hover,
.c1:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c1:hover::before,
.c1:focus::before {
  opacity: 1;
}

<body>
    <div>
      <div
        class="c0"
      >
        <a
          class="c1"
        >
          Buscar
        </a>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: grid;
  place-items: center;
}

.c1 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c1::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c1:hover,
.c1:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c1:hover::before,
.c1:focus::before {
  opacity: 1;
}

<div>
    <div
      class="c0"
    >
      <a
        class="c1"
      >
        Buscar
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Button Testing should render snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: grid;
  place-items: center;
}

.c1 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c1::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c1:hover,
.c1:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c1:hover::before,
.c1:focus::before {
  opacity: 1;
}

<body>
    <div>
      <div
        class="c0"
      >
        <a
          class="c1"
        >
          Buscar
        </a>
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: grid;
  place-items: center;
}

.c1 {
  font-size: 2rem;
  cursor: pointer;
  position: relative;
  padding: 0.25em 1em;
  border-radius: 0.25em;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EEEEEE;
  border: #EEEEEE 0.125em solid;
  text-shadow: 0 0 0.125em hsl(0 0% 100% / 0.3), 0 0 0.45em #EEEEEE;
  box-shadow: inset 0 0 0.5em 0 white,0 0 0.5em 0 white;
}

.c1::before {
  content: '';
  position: absolute;
  pointer-events: none;
  top: 120%;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.7;
  -webkit-filter: blur(1em);
  filter: blur(1em);
  background-color: #EEEEEE;
  -webkit-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  -ms-transform: perspective(1em) rotateX(40deg) scale(1,0.35);
  transform: perspective(1em) rotateX(40deg) scale(1,0.35);
}

.c1:hover,
.c1:focus {
  color: #393E46;
  background-color: #EEEEEE;
  text-shadow: none;
}

.c1:hover::before,
.c1:focus::before {
  opacity: 1;
}

<div>
    <div
      class="c0"
    >
      <a
        class="c1"
      >
        Buscar
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
